eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -121,10 +119,117 @@ function Confirm-ChangeLogEntry {
   return $true
 }
 
-function Set-TestChangeLog($TestVersion, $changeLogFile, $ReleaseEntry) {
-  Set-Content -Path $changeLogFile -Value @"
-# Release History
-## $TestVersion ($(Get-Date -f "yyyy-MM-dd"))
-- $ReleaseEntry
-"@
+function New-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Version,
+    [String]$Status="(Unreleased)",
+    [String]$Title,
+    [String[]]$Content
+  )
+
+  # Validate relase Status
+  $Status = $Status.Trim().Trim("()")
+  if ($Status -ne "Unreleased") {
+    $dateFormat = "yyyy-MM-dd"
+    $provider = [System.Globalization.CultureInfo]::InvariantCulture
+    try {
+      $Status = ([System.DateTime]::ParseExact($Status, $dateFormat, $provider)).ToString($dateFormat)
+    }
+    catch {
+        LogWarning "Invalid date [ $Status ] passed as status for Version [$Version]. Please use a valid date in the format '$dateFormat' or use '(Unreleased)'"
+    }
+  }
+
+  $Status = "($Status)"
+  if (!$Content) { $Content = @() }
+  if (!$Title) { $Title = "## $Version $Status" }
+
+  $newChangeLogEntry = [pscustomobject]@{ 
+    ReleaseVersion = $Version
+    ReleaseStatus  = $Status
+    ReleaseTitle   = $Title
+    ReleaseContent = $Content
+  }
+
+  return $newChangeLogEntry
+}
+
+function Add-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $false)]
+    [Hashtable]$ChangeLogEntries,
+    [Parameter(Mandatory = $true)]
+    [String]$ChangeLogLocation,
+    [Parameter(Mandatory = $true)]
+    [String]$NewEntryVersion,
+    [String]$NewEntryReleaseStatus="(Unreleased)",
+    [String[]]$NewEntryContent=""
+  )
+
+  if ($ChangeLogEntries.Contains($NewEntryVersion))
+  {
+    LogError "This Changelog already contains this entry. Do you mean to edit the entry?"
+    return
+  }
+
+  $newChangeLogEntry = New-ChangeLogEntry -Version $NewEntryVersion -Status $NewEntryReleaseStatus `
+  -Content $NewEntryContent
+  $ChangeLogEntries.Add($NewEntryVersion,$newChangeLogEntry)
+
+  $ChangeLogEntries
+  
+  Set-ChangeLogContent -ChangeLogLocation $ChangeLogLocation -ChangeLogEntries $ChangeLogEntries
+}
+
+function Edit-ChangeLogEntry {
```

Comment:
What value does this function really add? If you already have the changelog entry it seems pretty easy to set these value without needing to call this function.

Reply:
Well it also calls `Set-ChangeLogContent` which provides a way to write the Hashtable of changelog Entries back to the ChangeLog file.